org.eclipse.xtext.ui.shared/src/org/eclipse/xtext/ui/shared/internal/ListenerRegistrar.java

```diff
@@ -41,9 +46,25 @@ public void initialize() {
 		// after the fact, the project-open event is already done. Thus no build is triggered for the newly
 		// opened project and the index state is corrupt.
 		// thus we trigger a recovery build here
-		if (builderState.isEmpty()) {
-			buildManager.scheduleBuildIfNecessary(Arrays.asList(workspace.getRoot().getProjects()), IBuildFlag.RECOVERY_BUILD);
+		class RecoveryBuildTrigger extends WorkspaceJob {
+
+			public RecoveryBuildTrigger() {
+				super("Schedule xtext recovery build on start-up");
```

Comment:
Minor: `xtext` should be `Xtext` in the UI.

Reply:
You mean capital letter? How do I amend the PR?

Reply:
you can ammend your commit and force push the branch (am not sure this works with master though)